airflow/contrib/operators/bigquery_get_data.py

```diff
@@ -0,0 +1,115 @@
+
+    Example:
+
+    get_data = BigQueryGetDataOperator(
+        task_id='get_data_from_bq',
+        dataset_id='test_dataset',
+        table_id='Transaction_partitions',
+        max_results='100',
+        # selected_fields='DATE',
+        bigquery_conn_id='airflow-service-account'
+    )
+
+    :param dataset_id: The dataset ID of the requested table.
+    :type destination_dataset_table: string
+    :param table_id: The table ID of the requested table.
+    :type table_id: string
+    :param max_results: The maximum number of records (rows) to be fetched
+        from the table.
+    :type max_results: string
+    :param selected_fields: List of fields to return (comma-separated). If
+        unspecified, all fields are returned.
+    :type selected_fields: string
+    :param bigquery_conn_id: reference to a specific BigQuery hook.
+    :type bigquery_conn_id: string
+    :param delegate_to: The account to impersonate, if any.
+        For this to work, the service account making the request must have domain-wide
+        delegation enabled.
+    :type delegate_to: string
+
+    """
+    template_fields = ('dataset_id', 'table_id', 'max_results')
+    ui_color = '#e4f0e8'
+
+    @apply_defaults
+    def __init__(self,
+                 dataset_id,
+                 table_id,
+                 max_results='100',
+                 selected_fields=None,
+                 bigquery_conn_id='bigquery_default',
+                 delegate_to=None,
+                 *args,
+                 **kwargs):
+        super(BigQueryGetDataOperator, self).__init__(*args, **kwargs)
+        self.dataset_id = dataset_id
+        self.table_id = table_id
+        self.max_results = max_results
+        self.selected_fields = selected_fields
+        self.bigquery_conn_id = bigquery_conn_id
+        self.delegate_to = delegate_to
+
+    def execute(self, context):
+        logging.info('Fetching Data from:')
+        logging.info('Dataset: %s ; Table: %s ; Max Results: %s',
+                     self.dataset_id, self.table_id, self.max_results)
+
+        hook = BigQueryHook(bigquery_conn_id=self.bigquery_conn_id,
+                            delegate_to=self.delegate_to)
+
+        conn = hook.get_conn()
+        cursor = conn.cursor()
+        response = cursor.get_tabledata(dataset_id=self.dataset_id,
+                                        table_id=self.table_id,
+                                        max_results=self.max_results,
+                                        selected_fields=self.selected_fields)
+
+        logging.info('Total Extracted rows: %s', response['totalRows'])
+        rows = response['rows']
+
+        table_data = []
+        for dict_row in rows:
+            single_row = []
+            for fields in dict_row['f']:
+                single_row.append(fields['v'])
+            table_data.append(single_row)
+
+        logging.info('Sample Table Data [Top 2 rows]: %s', table_data[0:2])
```

Comment:
This needs to be removed. Logging the data is a potential security violation as it could potentially leak sensitive data from BigQuery into the logging infrastructure.